packages/lesswrong/components/users/DialogueMatchingPage.tsx

```diff
@@ -56,29 +70,97 @@ export type TagWithCommentCount = {
   commentCount: number
 }
 
-const styles = (theme: ThemeType): JssStyles => ({
+interface CommonDialogueUserRowProps {
+  checkId: string;
+  userIsChecked: boolean;
+  userIsMatched: boolean;
+  currentUser: UsersCurrent;
+  showBio: boolean | undefined;
+  showFrequentCommentedTopics: boolean | undefined;
+  showPostsYouveRead: boolean | undefined;
+}
+
+type DialogueUserRowProps<V extends boolean> = V extends true ? (CommonDialogueUserRowProps & {
+  targetUser: UpvotedUser;
+  showKarma: boolean;
+  showAgreement: boolean;
+}) : (CommonDialogueUserRowProps & {
+  targetUser: Omit<UsersOptedInToDialogueFacilitation, 'karma'>;
+  showKarma: false;
+  showAgreement: false;
+});
+
+
+type RowUser = UsersOptedInToDialogueFacilitation & {
+  [k in keyof Omit<UpvotedUser, '_id' | 'username' | 'displayName'>]?: never;
+};
+
+type CommonUserTableProps = {
+  classes: ClassesType<typeof styles>;
+  gridClassName: string,
+  currentUser: UsersCurrent;
+  userDialogueChecks: DialogueCheckInfo[];
+  showBio: boolean;
+  showPostsYouveRead: boolean;
+  showFrequentCommentedTopics: boolean;
+}
+
+type UserTableProps<V extends boolean> = V extends false ? (CommonUserTableProps & {
+  users: RowUser[];
+  showKarma: false;
+  showAgreement: false;
+  isUpvotedUser: false;
+  showHeaders: boolean;
+}) : (CommonUserTableProps & {
+  users: UpvotedUser[];
+  showKarma: boolean;
+  showAgreement: boolean;
+  isUpvotedUser: true;
+  showHeaders: boolean;
+});
+
+type NextStepsDialogProps = {
+  onClose: () => void;
+  userId: string;
+  targetUserId: string;
+  targetUserDisplayName: string;
+  dialogueCheckId: string;
+  classes: ClassesType<typeof styles>;
+};
+
+type MatchDialogueButtonProps = {
+  isMatched: boolean;
+  checkId: string,
+  targetUserId: string;
+  targetUserDisplayName: string;
+  currentUser: UsersCurrent;
+  classes: ClassesType<typeof styles>;
+};
+
+const styles = (theme: ThemeType) => ({
   root: {
     padding: 20,
     ...commentBodyStyles(theme),
+    background: theme.palette.background.default,
   },
   matchContainer: {
     maxWidth: 1300,
     padding: 20,
-    backgroundColor: theme.palette.grey[100],
+    background: theme.palette.panelBackground.default,
     borderRadius: 5,
   },
   matchContainerGridV1: {
-    display: 'grid',    //        checkbox         name         message                match                           upvotes                  agreement           tags    posts read   
-    gridTemplateColumns: `minmax(min-content, 60px) 100px minmax(min-content, 80px) minmax(min-content, 300px) minmax(min-content, 45px) minmax(min-content, 80px)  200px     550px`,
-    gridRowGap: '5px',
-    columnGap: '10px',
+    display: 'grid',    //      checkbox       name         message                match                 upvotes        agreement         tags    posts read
```

Comment:
I see V1 and V2, clean up when you're done deciding

Reply:
They're both used, for different kinds of tables.